src/Config.php

```diff
@@ -0,0 +1,208 @@
+<?php
+
+/*
+ * This file is part of the Flintstone package.
+ *
+ * (c) Jason M <emailfire@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE.md
+ * file that was distributed with this source code.
+ */
+
+namespace Flintstone;
+
+use Doctrine\Common\Cache\Cache;
+use Doctrine\Common\Cache\ArrayCache;
+use Doctrine\Common\Cache\ClearableCache;
+use Flintstone\Formatter\FormatterInterface;
+use Flintstone\Formatter\SerializeFormatter;
+
+class Config
+{
+    /**
+     * Default config.
+     *
+     * @var array
+     */
+    protected $config = array(
+        'dir' => '',
+        'ext' => '.dat',
+        'gzip' => false,
+        'cache' => true,
+        'formatter' => null,
+        'swap_memory_limit' => 2097152,    // 2MB
+    );
+
+    /**
+     * Constructor.
+     *
+     * @param array $config
+     */
+    public function __construct(array $config = array())
+    {
```

Comment:
You should normalize the default `$config` array with `array_replace` or `array_merge` which would remove any need for the `if` this would include adding a `$defaultConfig` protected property IMHO